### ISSUE TITLE
Add a `list` subcommand for printing the contents of a compressed file or archive

### ISSUE THREAD

**Issue by vrmiguel** [ouch-org/ouch#6]

**Comment by VuiMuich** [comment]
Hi first of all congrats, I really like this project!

IMO this one could be higher priority, as it would be very helpful in daily use.
I like to check archives bevor decompressing, whether they have subfolders or all the files in archive-root and depending on that set an output-path or not.

As a secondary enhancement `-t | --tree` and `-d | --depth` flags for the `list` subcommand might be nice additions.

**Comment by marcospb19** [comment]
@VuiMuich originally we planned on having `--tree` enable by default, however, that's not friendly for piping into another program or using as arguments in scripts.

So maybe we should have a way to output it without the extra tree formatting, what do you think? Do you have any flag suggestions?

**Comment by VuiMuich** [comment]
Oh, this totaly makes sense, piping a visually appealing tree is of little to no use :smile: 

In the spirit of #48 I was thinking of something like this:
```
$ ouch list example.zip
example.zip/File1
example.zip/Folder1
example.zip/Foler1/AnotherFile1
example.zip/Folder2
example.zip/Folder2/SubFolder1
example.zip/Folder2/Subfolder1/ThisFileIsDeep

$ ouch list --tree example.zip
example.zip
   L   File1
   L   Folder1
   |   L   AnotherFile1
   L   Folder2
       L   Subfolder1
           L   ThisFileIsDeep

$ ouch list --depth 1 example.zip
example.zip
./File1
./Folder1
./Foler1/AnotherFile1
./Folder2
./Folder2/SubFolder1

$ ouch list -t -d 1 example.zip
example.zip
   L   File1
   L   Folder1
   |   L   AnotherFile1
   L   Folder2
       L   Subfolder1

ouch list --files-only example.zip
$ ouch list example.zip
example.zip/File1
example.zip/Foler1/AnotherFile1
example.zip/Folder2/Subfolder1/ThisFileIsDeep
```
While writing this the idea of last example came to mind to have flags for `--files-only`  and `--dirs-only`

**Comment by marcospb19** [comment]
Implemented in #129.
